localstack/services/sns/resource_providers/aws_sns_subscription.py

```diff
@@ -128,6 +125,31 @@ def update(
         """
         Update a resource
 
-
         """
-        raise NotImplementedError
+        model = request.desired_state
+        model["Id"] = request.previous_state["Id"]
+        sns = request.aws_client_factory.sns
+
+        attrs = [
+            "DeliveryPolicy",
+            "FilterPolicy",
+            "FilterPolicyScope",
+            "RawMessageDelivery",
+            "RedrivePolicy",
+        ]
+        for a in attrs:
+            if a in model:
+                sns.set_subscription_attributes(
+                    SubscriptionArn=model["Id"],
+                    AttributeName=a,
+                    AttributeValue=self.attr_val(model[a]),
+                )
+        return ProgressEvent(
+            status=OperationStatus.SUCCESS,
+            resource_model=model,
+            custom_context=request.custom_context,
+        )
+
+    @staticmethod
+    def attr_val(val):
+        return json.dumps(val) if isinstance(val, dict) else str(val)
```

Comment:
it might be useful to also properly cast boolean values into lower cased string here with `localstack.utils.strings.canonicalize_bool_to_str`if the val is `bool`? 
Something like the following?
```suggestion
        if isinstance(val, dict):
            return json.dumps(val)
        elif isinstance(val, bool):
            return canonicalize_bool_to_str(val)
        else:
            return str(val)
```

Reply:
in the resource providers we avoid importing outside of cloudformation since in the future the resource providers will be allocated in a new repo

Reply:
I see! We should use the same logic as the helper though in the resource provider, because it's the right behavior to lower case boolean values I think for SNS